From 5d3deb50b4dd6985372eae0a306e7414172e2434 Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Wed, 19 Nov 2014 16:14:32 -0500
Subject: [ethernet] r8169: fix the default setting of rx vlan

Message-id: <1416413687-12219-2-git-send-email-ivecera@redhat.com>
Patchwork-id: 100351
O-Subject: [RHEL7.1 PATCH 01/16] r8169: fix the default setting of rx vlan
Bugzilla: 1165764
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

BZ: #1165764

Upstream commit(s):
commit 48c20407f49039d8da155ca00877ec5f057799e6
Author: hayeswang <hayeswang@realtek.com>
Date:   Fri Sep 12 11:35:11 2014 +0800

    r8169: fix the default setting of rx vlan

    If the parameter "features" of __rtl8169_set_features() is equal to
    dev->features, the variable "changed" is alwayes 0, and nothing would
    be changed.

    Signed-off-by: Hayes Wang <hayeswang@realtek.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/realtek/r8169.c b/drivers/net/ethernet/realtek/r8169.c
index 30217c1..d54e6c8 100644
--- a/drivers/net/ethernet/realtek/r8169.c
+++ b/drivers/net/ethernet/realtek/r8169.c
@@ -7116,7 +7116,12 @@ static int rtl_open(struct net_device *dev)
 
  rtl8169_init_phy(dev, tp);
 
- __rtl8169_set_features(dev, dev->features);
+ if (dev->features & NETIF_F_HW_VLAN_CTAG_RX)
+  tp->cp_cmd |= RxVlan;
+ else
+  tp->cp_cmd &= ~RxVlan;
+
+ RTL_W16(CPlusCmd, tp->cp_cmd);
 
  rtl_pll_power_up(tp);
 
-- 
1.7.1